### DISCUSSION TITLE
block cloning save to use?

### DISCUSSION THREAD

**Discussion by mabod** [openzfs/zfs#16886] [General]
Is block cloning save to use with zfs 2.2.7? I see in the source code   `./module/zfs/zfs_vnops.c`
```
  61 /*
  62  * Enable the experimental block cloning feature.  If this setting is 0, then
  63  * even if feature@block_cloning is enabled, attempts to clone blocks will act
  64  * as though the feature is disabled.
  65  */
  66 int zfs_bclone_enabled = 0;
```

Does that mean it is not fixed? What about zfs 2.3.0?

**Comment by amotin** [comment]
In 2.3.0 it is enabled by default.  In 2.2.7 it is disabled mostly out of caution.  It should work fine there, though in 2.3 I've optimized it a lot.
